internal/services/netapp/netapp_volume_helper.go

```diff
@@ -0,0 +1,978 @@
+		replicationObject.EndpointType = &endpointType
+	}
+	if v, ok := replicationRaw["remote_volume_location"]; ok {
+		replicationObject.RemoteVolumeRegion = utils.String(v.(string))
+	}
+	if v, ok := replicationRaw["remote_volume_resource_id"]; ok {
+		replicationObject.RemoteVolumeResourceId = v.(string)
+	}
+	if v, ok := replicationRaw["replication_frequency"]; ok {
+		replicationSchedule := volumes.ReplicationSchedule(translateTFSchedule(v.(string)))
+		replicationObject.ReplicationSchedule = &replicationSchedule
+	}
+
+	return &volumes.VolumePropertiesDataProtection{
+		Replication: &replicationObject,
+	}
+}
+
+func expandNetAppVolumeDataProtectionSnapshotPolicy(input []interface{}) *volumes.VolumePropertiesDataProtection {
+	if len(input) == 0 || input[0] == nil {
+		return &volumes.VolumePropertiesDataProtection{}
+	}
+
+	snapshotObject := volumes.VolumeSnapshotProperties{}
+
+	snapshotRaw := input[0].(map[string]interface{})
+
+	if v, ok := snapshotRaw["snapshot_policy_id"]; ok {
+		snapshotObject.SnapshotPolicyId = utils.String(v.(string))
+	}
+
+	return &volumes.VolumePropertiesDataProtection{
+		Snapshot: &snapshotObject,
+	}
+}
+
+func expandNetAppVolumeDataProtectionSnapshotPolicyPatch(input []interface{}) *volumes.VolumePatchPropertiesDataProtection {
+	if len(input) == 0 || input[0] == nil {
+		return &volumes.VolumePatchPropertiesDataProtection{}
+	}
+
+	snapshotObject := volumes.VolumeSnapshotProperties{}
+
+	snapshotRaw := input[0].(map[string]interface{})
+
+	if v, ok := snapshotRaw["snapshot_policy_id"]; ok {
+		snapshotObject.SnapshotPolicyId = utils.String(v.(string))
+	}
+
+	return &volumes.VolumePatchPropertiesDataProtection{
+		Snapshot: &snapshotObject,
+	}
+}
+
+func flattenNetAppVolumeGroupVolumes(ctx context.Context, input *[]volumegroups.VolumeGroupVolumeProperties, metadata sdk.ResourceMetaData) ([]NetAppVolumeGroupVolume, error) {
+	results := make([]NetAppVolumeGroupVolume, 0)
+
+	if len(*input) == 0 || input == nil {
+		return results, fmt.Errorf("received empty volumegroups.VolumeGroupVolumeProperties slice")
+	}
+
+	for _, item := range *input {
+		volumeGroupVolume := NetAppVolumeGroupVolume{}
+
+		props := item.Properties
+
+		volumeGroupVolume.Name = getResourceNameString(item.Name)
+		volumeGroupVolume.VolumePath = props.CreationToken
+		volumeGroupVolume.ServiceLevel = string(*props.ServiceLevel)
+		volumeGroupVolume.SubnetId = props.SubnetId
+		volumeGroupVolume.CapacityPoolId = utils.NormalizeNilableString(props.CapacityPoolResourceId)
+		volumeGroupVolume.Protocols = *props.ProtocolTypes
+		volumeGroupVolume.SecurityStyle = string(*props.SecurityStyle)
+		volumeGroupVolume.SnapshotDirectoryVisible = *props.SnapshotDirectoryVisible
+		volumeGroupVolume.ThroughputInMibps = *props.ThroughputMibps
+		volumeGroupVolume.Tags = *item.Tags
+		volumeGroupVolume.ProximityPlacementGroupId = utils.NormalizeNilableString(props.ProximityPlacementGroup)
+		volumeGroupVolume.VolumeSpecName = *props.VolumeSpecName
```

Comment:
```suggestion
		volumeGroupVolume.VolumeSpecName = pointer.From*props.VolumeSpecName)
```

Reply:
There is a missing open parenthesis. I'll do this locally and push.